### ISSUE TITLE
Java API for querying TSDB with millisecond resolution

### ISSUE THREAD

**Issue by spicoflorin** [OpenTSDB/opentsdb#635]
While debugging the TSDMain class, I found out that the filtering at millisecond is not performed at the database level, but in client code in HTTPJonSerializer. I believe that this is the current supported implementation. 
Can we have a Java API that is able to query the TSDB with the millisecond precision at the database level?
Thanks.

**Comment by manolama** [comment]
Yeah I want to try and get that into 2.3. It requires a bit of munging to implement but shouldn't be too difficult.
